packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx

```diff
@@ -88,9 +88,9 @@ const VisualizationCard: FC = memo(() => {
     }
 
     const getCsvLink = async (csvLimit: number | null, onlyRaw: boolean) => {
-        if (explore?.name && unsavedChartVersion?.metricQuery && projectUuid) {
+        if (explore?.name && unsavedChartVersion?.metricQuery) {
             const csvResponse = await downloadCsv({
-                projectUuid,
+                projectUuid: projectUuid || unsavedChartVersion.projectUuid,
```

Comment:
you can get a fallback projectUuid from the URI by using
```
const { projectUuid } = useParams<{ projectUuid: string }>();
```
I would add this chunk of code to `Explore` component rather than `VisualizationCard` and pass it down as a prop.

then you can use `savedChart.projectUuid || projectUuid` in here

LMK if you have questions!